### PR TITLE
Add missing function getsocketoption

### PR DESCRIPTION
**Pull request by rtpmsys**
WiFimanager.cpp connect function truncated all connnect timeout values below 1000ms to 0ms and rounded timeouts to the full second, e.g. 2.5 seconds became 2 seconds. For timeouts below 1000ms,  the connect function failed. This fix handles timeouts correctly, by putting the fractional part of the timeout into the timeout_us variable.

Added function getsocketoptions, it was missing.

### CODE REVIEW

**Comment by me-no-dev** [comment]
This PR seems to include two non-relevant changes. Could you please split the timeout fix and `getSocketOption` into separate PRs?

**Comment by TD-er** [comment]
Why keep on insisting to have a `WiFiClient::setTimeout` function in this class using seconds while every other piece of code uses msec?
Cast it to `Client` and call that timeout will expect msec.

**Comment by rtpmsys** [comment]
I submitted only the timeout correction and omitted the 

> This PR seems to include two non-relevant changes. Could you please split the timeout fix and `getSocketOption` into separate PRs?

I submitted just the timeout change which fixes the error with connect timeout values below 1 second.

**Comment by rtpmsys** [comment]
> Any updates @rtpmsys?

I submitted a new pull request with just the timeout fix. Hope it makes it through the system, now.

**Comment by rtpmsys** [comment]
> Why keep on insisting to have a `WiFiClient::setTimeout` function in this class using seconds while every other piece of code uses msec?
> Cast it to `Client` and call that timeout will expect msec., which is a clear indication this design choice in the past was wrong.

You are right that the setTimeout function only expects seconds, I guess for compatibility reasons? The setTimeout luckily only seems to affect the write and read from the socket, not the connect.
My proposed fix deals with the connect timeout, which comes before the write or read, therefore it is much more important that it can be made smaller than 1 second.

**Comment by mrengineer7777** [comment]
@rtpmsys #7686 was merged.  Time to update this PR?

File: libraries/WiFi/src/WiFiClient.cpp [hunk 1]
```diff
@@ -231,7 +231,7 @@ int WiFiClient::connect(IPAddress ip, uint16_t port, int32_t timeout)
     FD_ZERO(&fdset);
     FD_SET(sockfd, &fdset);
     tv.tv_sec = _timeout / 1000;
-    tv.tv_usec = 0;
```
Comment (rtpmsys) [inline libraries/WiFi/src/WiFiClient.cpp:234]: Moved into another PR

**Comment by rtpmsys** [comment]
I will submit a new PR for adding the missing function getSocketOption.
